src/renderer/uia/UiaRenderer.cpp

```diff
@@ -180,15 +209,23 @@ UiaEngine::UiaEngine() noexcept :
 // - S_OK, else an appropriate HRESULT for failing to allocate or write.
 [[nodiscard]] HRESULT UiaEngine::EndPaint() noexcept
 {
+    RETURN_HR_IF(S_FALSE, !_isEnabled);
     RETURN_HR_IF(E_INVALIDARG, !_isPainting); // invalid to end paint when we're not painting
 
-    if (_isEnabled)
+    // Fire UIA Events here
+    if (_selectionChanged)
     {
-        _isPainting = false;
-
-        // fire UIA events here
+        try
+        {
+            _dispatcher->SignalSelectionChanged();
+        }
+        CATCH_LOG();
     }
 
+    _selectionChanged = false;
+    _prevSelection.clear();

```

Comment:
I'm mildly worried that while there's a selection, we'll signal events every frame saying "there's a selection". Should we only signal a selection when it changes from one frame to the next? Or is this right? (I don't know how to tests this manually myself, so feel free to resolve if this isn't an issue).

Reply:
Not a problem. We're tracking when a selection _changes_ not _exists_. So we're only signaling in a frame _iff_ the selection changed from the last one to this one.